fgrpc/grpcrunner.go

```diff
@@ -35,9 +35,11 @@ import (
 
 const (
 	// DefaultGRPCPort is the Fortio gRPC server default port number.
-	DefaultGRPCPort = "8079"
-	prefixHTTP      = "http://"
-	prefixHTTPS     = "https://"
+	DefaultGRPCPort  = "8079"
```

Comment:
Did you check that
```
fortio load -grpc localhost
# and
fortio grpcping localhost
```
still works (and goes to localhost:8079)

Reply:
Yes. See the following details:

__Start Fortio Server__
```
$ docker run --rm --name fortio -p 8080:80 -p 8079:8079 istio/fortio:webtest server -loglevel debug -http-port 80
17:36:12 I logger.go:97> Log level is now 0 Debug (was 2 Info)
17:36:12 V stats.go:517> Will use [50 75 90 99 99.9] for percentiles
Fortio 0.8.1-pre grpc 'ping' server listening on [::]:8079
Fortio 0.8.1-pre https redirector server listening on [::]:8081
Fortio 0.8.1-pre echo server listening on [::]:80
UI started - visit:
http://localhost:80/fortio/   (or any host/ip reachable on this server)
17:36:12 V uihandler.go:96> Using resources directory set at link time: /usr/local/lib/fortio
17:36:12 I fortio_main.go:186> All fortio 0.8.1-pre 2018-03-26 22:14 6aa57194a7426326cdbc9c164753c96e67334242-dirty go1.10 servers started!
```

__Run `load -grpc localhost` command__
```
$ docker exec fortio /usr/local/bin/fortio grpcping localhost
17:37:07 I grpcrunner.go:199> grpc destination: localhost
17:37:07 I grpcrunner.go:227> grpc destination set to: localhost:8079
Clock skew histogram usec : count 1 avg -22.923 +/- 0 min -22.923 max -22.923 sum -22.923
# range, mid point, percentile, count
>= -22.923 <= -22.923 , -22.923 , 100.00, 1
# target 50% -22.923
RTT histogram usec : count 3 avg 756.864 +/- 84.46 min 648.227 max 854.181 sum 2270.592
# range, mid point, percentile, count
>= 648.227 <= 700 , 674.113 , 33.33, 1
> 700 <= 800 , 750 , 66.67, 1
> 800 <= 854.181 , 827.091 , 100.00, 1
# target 50% 750
17:37:07 I pingsrv.go:117> Ping RTT 756864 (avg of 648227, 768184, 854181 ns) clock skew -22923
```

__Run `grpcping localhost` command__
```
$ docker exec fortio /usr/local/bin/fortio load -grpc localhost
Fortio 0.8.1-pre running at 8 queries per second, 4->4 procs, for 5s: localhost
17:37:40 I grpcrunner.go:99> Starting grpc test for localhost with 4 threads at 8.0 qps
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
Starting at 8 qps with 4 thread(s) [gomax 4] for 5s : 10 calls each (total 40)
17:37:45 I periodic.go:530> T000 ended after 5.005335345s : 10 calls. qps=1.997868136845085
17:37:45 I periodic.go:530> T002 ended after 5.005375398s : 10 calls. qps=1.9978521499098159
17:37:45 I periodic.go:530> T001 ended after 5.005953009s : 10 calls. qps=1.9976216280938726
17:37:45 I periodic.go:530> T003 ended after 5.006542006s : 10 calls. qps=1.997386616953514
Ended after 5.006640388s : 40 calls. qps=7.9894
Sleep times : count 36 avg 0.55170978 +/- 0.001628 min 0.548772401 max 0.554760104 sum 19.8615522
Aggregated Function Time : count 40 avg 0.001034316 +/- 0.000333 min 0.000373654 max 0.001956965 sum 0.041372639
# range, mid point, percentile, count
>= 0.000373654 <= 0.001 , 0.000686827 , 37.50, 15
> 0.001 <= 0.00195697 , 0.00147848 , 100.00, 25
# target 50% 0.00119139
# target 75% 0.00157418
# target 90% 0.00180385
# target 99% 0.00194165
# target 99.9% 0.00195543
Health SERVING : 40
All done 40 calls (plus 4 warmup) 1.034 ms avg, 8.0 qps
```

__Observe Server Logs__
```
17:36:51 V pingsrv.go:43> Ping called {Seq:0 Ts:0 Payload:} (ctx context.Background.WithCancel.WithCancel.WithValue(peer.peerKey{}, &peer.Peer{Addr:(*net.TCPAddr)(0xc420163950), AuthInfo:credentials.AuthInfo(nil)}).WithValue(transport.streamKey{}, <stream: 0xc4201d4500, /fgrpc.PingServer/Ping>).WithValue(metadata.mdIncomingKey{}, metadata.MD{":authority":[]string{"localhost:8079"}, "user-agent":[]string{"grpc-go/1.9.2"}}))
<SNIP>
17:37:07 V pingsrv.go:43> Ping called {Seq:0 Ts:0 Payload:} (ctx context.Background.WithCancel.WithCancel.WithValue(peer.peerKey{}, &peer.Peer{Addr:(*net.TCPAddr)(0xc420156780), AuthInfo:credentials.AuthInfo(nil)}).WithValue(transport.streamKey{}, <stream: 0xc4201d48c0, /fgrpc.PingServer/Ping>).WithValue(metadata.mdIncomingKey{}, metadata.MD{":authority":[]string{"localhost:8079"}, "user-agent":[]string{"grpc-go/1.9.2"}}))
<SNIP>
```

Reply:
Adding test cases to `Webtest.sh`.

Reply:
as you can see
```
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
17:37:40 I grpcrunner.go:199> grpc destination: localhost
17:37:40 I grpcrunner.go:227> grpc destination set to: localhost:8079
```
is way too much logging

Reply:
The normalization is happing once per thread b/c `grpcDestination` is being called by the `Dial` func, which gets called once per thread in `RunGRPCTest`. I think the log settings I added simply expose an existing issue. Can you create an Issue, assign me and I'll refactor in a separate PR?